allennlp/training/metrics/srl_eval_scorer.py

```diff
@@ -0,0 +1,157 @@
+            raise ConfigurationError(f"srl-eval.pl not found at {self._srl_eval_path}.")
+        tempdir = tempfile.mkdtemp()
+        gold_path = os.path.join(tempdir, "gold.txt")
+        predicted_path = os.path.join(tempdir, "predicted.txt")
+
+        with open(predicted_path, "w") as predicted_file, open(gold_path, "w") as gold_file:
+            for verb_index, sentence, predicted_tag_sequence, gold_tag_sequence in zip(
+                    batch_verb_indices,
+                    batch_sentences,
+                    batch_conll_formatted_predicted_tags,
+                    batch_conll_formatted_gold_tags):
+                write_conll_formatted_tags_to_file(predicted_file,
+                                                   gold_file,
+                                                   verb_index,
+                                                   sentence,
+                                                   predicted_tag_sequence,
+                                                   gold_tag_sequence)
+        perl_script_command = ["perl", self._srl_eval_path, predicted_path, gold_path]
+        completed_process = subprocess.run(perl_script_command, stdout=subprocess.PIPE,
+                                           universal_newlines=True, check=True)
+        for line in completed_process.stdout.split("\n"):
+            stripped = line.strip().split()
+            if len(stripped) == 7:
+                tag = stripped[0]
+                # overall metrics are calculated in get_metric
+                if tag == "Overall":
+                    continue
+                # This line contains results for a span
+                num_correct = int(stripped[1])
+                num_excess = int(stripped[2])
+                num_missed = int(stripped[3])
+                self._true_positives[tag] += num_correct
+                self._false_positives[tag] += num_excess
+                self._false_negatives[tag] += num_missed
+        shutil.rmtree(tempdir)
+
+    def get_metric(self, reset: bool = False):
+        """
+        Returns
+        -------
+        A Dict per label containing following the span based metrics:
+        precision : float
+        recall : float
+        f1-measure : float
+
+        Additionally, an ``overall`` key is included, which provides the precision,
+        recall and f1-measure for all spans.
+        """
+        all_tags: Set[str] = set()
+        all_tags.update(self._true_positives.keys())
+        all_tags.update(self._false_positives.keys())
+        all_tags.update(self._false_negatives.keys())
+        all_metrics = {}
+        for tag in all_tags:
+            precision, recall, f1_measure = self._compute_metrics(self._true_positives[tag],
+                                                                  self._false_positives[tag],
+                                                                  self._false_negatives[tag])
+            precision_key = "precision" + "-" + tag
```

Comment:
Can the user possibly provide tags such that `tag == "overall"` or is that entirely decided by the Perl script? Please assert against it if needed.